src/dmd/expression.d

```diff
@@ -5163,6 +5163,19 @@ extern (C++) final class CastExp : UnaExp
         return to ? new CastExp(loc, e1.syntaxCopy(), to.syntaxCopy()) : new CastExp(loc, e1.syntaxCopy(), mod);
     }
 
+    override bool isLvalue()
+    {
+        //printf("e1.type = %s, to.type = %s\n", e1.type.toChars(), to.toChars());
+        return e1.type.mutableOf().unSharedOf() == to.mutableOf().unSharedOf();
```

Comment:
Tests look good now, but shouldn't this use `implicitConvTo` to satisfy "cast(U) expressions applied to lvalues of type T when T* is implicitly convertible to U*;"?